scripts/resolve-includes.groovy

```diff
@@ -0,0 +1,140 @@
+        def stderrLogFile = "mvn-std-err.log"
+        log.error("Maven build unsuccessful with code ${mvnExec.exitValue()}, " +
+                "see ${Path.of(logdir).resolve(stderrLogFile).toString()}.")
+        logDebug(stderrLogFile, err.toString().lines().collect(Collectors.toList()))
+        throw new RuntimeException("Mvn command ended with error code ${mvnExec.exitValue()}")
+    }
+    return out.toString().lines()
+}
+
+/**
+ * Trim path prefix up to given basedir.
+ * @param value absolute path to be trimmed
+ * @param prefix absolute path to be used as prefix
+ * @return relative path between the two
+ */
+private String trimPrefix(Path value, Path prefix) {
+    return prefix.relativize(value).toString();
+}
+
+/**
+ * Return a list of string paths to pom project directories that are supposed to be leaf ones - no sub-modules.
+ * @param projects string paths for all the projects within reactor
+ * @return list of strings for paths that don't have any children in the list
+ */
+private List<String> resolveLeafProjects(Stream<String> projects) {
+    List<String> lines = projects.map({ it -> trimPrefix(Path.of(it), new File(basedir).toPath()) }).collect(Collectors.toList()).sort()
+    List<String> leafProjects = new ArrayList<>()
+    def logFile = "leaf-projects.log"
+    log.info("Starting leaf projects resolution, results will be in ${Path.of(logdir).resolve(logFile).toString()}")
+    for (int i = 0; i < lines.size(); i++) {
+        String current = lines[i]
+        if (lines.size() > i + 1) {
+            String next = lines[i + 1]
+            if (!Path.of(next).startsWith(current)) {
+                log.debug("Adding $current as a leaf project, it has no child")
+                leafProjects.add(current)
+            } else {
+                log.debug("Skipping '$current' project, cause it has child '$next'")
+            }
+        } else {
+            log.debug("Adding $current as a leaf project, cause no other project is in reactor.")
+            leafProjects.add(current)
+        }
+    }
+    logDebug(logFile, leafProjects)
+    return leafProjects
+}
+
+/**
+ * Log into given file contents of the provided list of paths.
+ * @param fileName target dir to be created within logdir (passed from config)
+ * @param projectList list of paths
+ */
+private void logDebug(String fileName, ArrayList<String> projectList) {
+    new File(logdir + "/" + fileName).write projectList.stream().map({ it ->
+        it + System.lineSeparator()
+    }).collect(Collectors.joining())
+}
+
+/**
+ * Recursively find all folders that contain pom.xml
+ * @return list of string paths
+ */
+private List<String> findAllDirsWithPom() {
+    File base = new File(basedir)
+    List<File> poms = new ArrayList<>()
+    base.traverse(type: groovy.io.FileType.FILES, nameFilter: "pom.xml") { it -> poms.add(it) }
+    return poms.stream().map({pom -> trimPrefix(pom.toPath().getParent().toAbsolutePath(), base.toPath())}).collect(Collectors.toList())
```

Comment:
kiegroup formatting format this line fo me as:
```
    return poms.stream().map({ pom -> trimPrefix(pom.toPath().getParent().toAbsolutePath(), base.toPath()) }).collect(Collectors.toList())
```

Reply:
My bad, will reformat using kiegroup template.